### PR TITLE
The 'too many scroll requests' exception should return 429 status

### PR DESCRIPTION
**Pull request by quux00**
The existing 500 error implies a temporary issue in the Elasticsearch server, which
is not the case. So a 429 is more appropriate to indicate that the user needs to
take some action to avoid getting the error again.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/SearchService.java [hunk 1]
```diff
@@ -952,7 +953,12 @@ final ReaderContext createAndPutReaderContext(
                             + "This limit can be set by changing the ["
                             + MAX_OPEN_SCROLL_CONTEXT.getKey()
                             + "] setting."
-                    );
+                    ) {
+                        @Override
+                        public RestStatus status() {
+                            return RestStatus.BAD_REQUEST;
+                        }
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/SearchService.java:960]: Do we think this is a `400` or a `429`?

Comment (quux00) [reply server/src/main/java/org/elasticsearch/search/SearchService.java:960]: Interesting question. I've always interpreted 429 "too many requests" as a rate. 
> The HTTP 429 Too Many Requests response status code indicates the user has sent too many requests in a given amount of time

This is not an amount of time rate. It is just a count that within a query there are too many scroll requests, which you can see by the if-guard around this section:

```
        if (source.scriptFields() != null && source.size() != 0) {
            int maxAllowedScriptFields = searchExecutionContext.getIndexSettings().getMaxScriptFields();
            if (source.scriptFields().size() > maxAllowedScriptFields) {
```

Comment (quux00) [reply server/src/main/java/org/elasticsearch/search/SearchService.java]: Got it. Changed.

**Review by cbuescher — APPROVED** [review]
With the change in the error code that was suggested this LGTM.

**Review by martijnvg — APPROVED** [review]
lgtm

**Review by benwtrent — APPROVED** [review]

**Pull request merged**